components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2387,6 +2395,10 @@ void RewardsServiceImpl::OnFilesDeletedForCompleteReset(
 }
 
 void RewardsServiceImpl::ExternalWalletConnected() {
+  // When the user connects an external wallet, turn off search result ads since
+  // users cannot earn BAT for them yet. The user can turn them on manually.
+  profile_->GetPrefs()->SetBoolean(brave_ads::prefs::kOptedInToSearchResultAds,
+                                   false);
```

Comment:
Sanity checked with @aseren and this logic is correct for desktop, android and iOS. iOS users are always disconnected. Thanks

Reply:
We can also confirm that if a user resets Rewards that this pref is cleared.